### PR TITLE
test(pci-vouchers): add tests and do some refacto

### PR DESCRIPTION
**Pull request by kqesar**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | DTCORE-2360
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

test(pci-vouchers): add tests and do some refacto

### CODE REVIEW

**Review by frenautvh — APPROVED** [review]

File: packages/manager/apps/pci-vouchers/src/components/vouchers/listing/Product.spec.tsx [hunk 1]
```diff
@@ -10,14 +10,12 @@ vi.mock('@ovhcloud/manager-components', async () => ({
...
-        changeLanguage: () => new Promise(() => {}),
-      },
-    };
-  },
+  useTranslation: () => ({
+    t: (str: string) => str,
+    i18n: {
+      changeLanguage: () => new Promise(() => {}),
+    },
+  }),
```
Comment (darsene) [inline packages/manager/apps/pci-vouchers/src/components/vouchers/listing/Product.spec.tsx:18]: already defined in `packages/manager/apps/pci-vouchers/src/setupTests.ts` so no need to define it one more time

File: packages/manager/apps/pci-vouchers/src/components/vouchers/listing/DisplayName.spec.tsx [hunk 2]
```diff
@@ -1,72 +1,68 @@
...
-        changeLanguage: () => new Promise(() => {}),
-      },
-    };
-  },
+  useTranslation: () => ({
+    t: (str: string) => str,
+    i18n: {
+      changeLanguage: () => new Promise(() => {}),
+    },
+  }),
```
Comment (darsene) [inline packages/manager/apps/pci-vouchers/src/components/vouchers/listing/DisplayName.spec.tsx:15]: already defined in packages/manager/apps/pci-vouchers/src/setupTests.ts so no need to define it one more time

File: packages/manager/apps/pci-vouchers/src/pages/AddVoucherPage.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}));
+
+vi.mock('@ovhcloud/manager-components', () => ({
+  useNotifications: () => vi.fn(),
+}));
+
+vi.mock('react-i18next', () => ({
+  useTranslation: () => ({ t: (key) => key }),
+}));
+
```
Comment (darsene) [inline packages/manager/apps/pci-vouchers/src/pages/AddVoucherPage.spec.tsx:22]: already defined in `packages/manager/apps/pci-vouchers/src/setupTests.ts` so no need to define it one more time

File: packages/manager/apps/pci-vouchers/src/components/vouchers/listing/Product.spec.tsx (hunk 1 above)
Comment (kqesar) [reply packages/manager/apps/pci-vouchers/src/components/vouchers/listing/Product.spec.tsx]: done

File: packages/manager/apps/pci-vouchers/src/components/vouchers/listing/DisplayName.spec.tsx (hunk 2 above)
Comment (kqesar) [reply packages/manager/apps/pci-vouchers/src/components/vouchers/listing/DisplayName.spec.tsx]: done

File: packages/manager/apps/pci-vouchers/src/pages/AddVoucherPage.spec.tsx (hunk 3 above)
Comment (kqesar) [reply packages/manager/apps/pci-vouchers/src/pages/AddVoucherPage.spec.tsx:22]: done

**Review by chipp972 — DISMISSED** [review]

**Review by MaximeBajeux — DISMISSED** [review]

**Review by chipp972 — DISMISSED** [review]

**Review by anooparveti — COMMENTED** [review]
Can you resolve the conflicts?

**Review by qpavy — APPROVED** [review]

**Review by frenautvh — APPROVED** [review]

**Review by sidlynx — DISMISSED** [review]

**Review by seven-amid — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**